common/src/main/java/com/wynntils/features/combat/RangeVisualizerFeature.java

```diff
@@ -0,0 +1,148 @@
+import com.mojang.blaze3d.vertex.BufferBuilder;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.features.Feature;
+import com.wynntils.mc.event.ItemInHandRenderEvent;
+import com.wynntils.mc.event.PlayerRenderEvent;
+import com.wynntils.models.gear.type.GearInfo;
+import com.wynntils.models.gear.type.GearMajorId;
+import com.wynntils.utils.mc.ComponentUtils;
+import com.wynntils.utils.mc.McUtils;
+import com.wynntils.utils.render.buffered.CustomRenderType;
+import net.minecraft.client.Camera;
+import net.minecraft.client.gui.screens.inventory.InventoryScreen;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderDispatcher;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Position;
+import net.minecraft.world.entity.player.Player;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.joml.Matrix3f;
+import org.joml.Matrix4f;
+
+import java.util.List;
+
+@ConfigCategory(Category.COMBAT)
+public class RangeVisualizerFeature extends Feature {
+
+    private static final MultiBufferSource.BufferSource BUFFER_SOURCE = MultiBufferSource.immediate(new BufferBuilder(256));
+    private static final int SEGMENTS = 128;
+    private static final float HEIGHT = 0.1f;
+
+    /**
+     * Deals with rendering the circle for other party members only.
+     */
+    @SubscribeEvent
+    public void onPlayerRender(PlayerRenderEvent e) {
+        if (!Models.Player.isLocalPlayer(e.getPlayer())) return; // Don't render for ghost/npc
+        String playerName = ComponentUtils.getUnformatted(e.getPlayer().getName());
+        if (ComponentUtils.getUnformatted(McUtils.player().getName()).equals(playerName)) {
+            if (McUtils.mc().screen instanceof InventoryScreen) return; // Don't render for preview in inventory
+        }
+        if (Models.Party.isInParty() && !Models.Party.getPartyMembers().contains(playerName)) return; // Other players must be in party
+
+        // We are getting the item info the same way as GearViewerScreen since we care about other people's items
+        String gearName = ComponentUtils.getUnformatted(e.getPlayer().getMainHandItem().getHoverName());
+        GearInfo gearInfo = Models.Gear.getGearInfoFromApiName(gearName);
+        if (gearInfo == null) return;
+        List<GearMajorId> majorIds = gearInfo.fixedStats().majorIds();
+        if (majorIds.isEmpty()) return;
+
+        // Major IDs that we can visualize:
+        // Taunt (12 blocks)
+        // Saviour's Sacrifice (8 blocks)?
+        // Heart of the Pack (8 blocks)?
+        // Guardian (8 blocks)?
+        // Marked with a ? needs additional confirmation
+
+        for (GearMajorId majorId : majorIds) {
+            if (majorId.name().equals("Taunt")) {
```

Comment:
I'd recommend you check the majorId.id() instead, which is like `TAUNT`. This is supposed to be more stable. In theory, the name can change (perhaps if we introduce i18n of gear?)